content/blog/blog-2020-07-10-cassandra-ga.md

```diff
@@ -0,0 +1,78 @@
+---
+date: 2020-07-10
+---
+
+# KUDO Cassandra 1.0.0 Released
+
+We have released (KUDO Cassandra)[https://github.com/mesosphere/kudo-cassandra-operator] 1.0.0 for general availability. 
+
+Some highlights from this release are:
+
+* Backup & Restore
+* Node Replace/Node Recovery
+* Multi-Datacenter Cluster
+
+<!-- more -->
+
+## Backup & Restore
+
+For KUDO Cassandra, we have integrated (Cassandra Medusa)[https://github.com/thelastpickle/cassandra-medusa] to allow easy backup and restore operations.
+Medusa is a backup system for Cassandra that provides support for backups to Amazon S3, Google Cloud Storage and local storage. At the moment we have only integrated the AWS S3 solution, but can easily add support for the other storage providers.
+
+With version 1.0.0, it is possible to do a full cluster backups and restore the backups to a new instance of the cluster - it is not possible to restore single nodes, but as Cassandra is a distributed database, a full cluster restore is a more probable use case.
+
+To enable backup and restore functionality, there is a single parameter: `BACKUP_RESTORE_ENABLED`. This will deploy a sidecar container on each Cassandra Node which contains Medusa. The actual backup is triggered by a Job that execs Medusa in the sidecar.
+
+For a restore, the parameter sets up an init container before the actual start of the container which downloads the backup and initialises the data for that node.
+
+Details about this feature can be found (here)[https://github.com/mesosphere/kudo-cassandra-operator/blob/master/docs/backup.md].
+
+## Node Eviction/Node Recovery
+
+One of the tasks of running a bigger Cassandra cluster is handling node failures and node replacement. There are generally two options for the storage provider. Either elastic block storage, which makes it easy to have the same data available on multiple servers, but limits the performance and is an additional complex software to manage. The second option is local storage, which provides the best performance, but has the risk of data loss in case of failure.
+
+The data loss is not really a problem for Cassandra - as long as the replication factor for a keyspace is setup correctly, copies of the data are stored on different nodes. But local storage requires manual intervention when a Cassandra node really fails. The cluster needs to be informed that a node was lost and was replaced, in case of Kubernetes the persistent volume and persistent volume claims need to be cleaned up, etc.
```

Comment:
```suggestion
Apache Cassandra is designed to tolerate node failure - as long as the minimum number of required nodes are up and running and the replication factor for a keyspace is setup correctly, copies of the data are stored on different nodes. But local storage requires manual intervention when a Cassandra node really fails. The cluster needs to be informed that a node was lost and was replaced, in case of Kubernetes the persistent volume and persistent volume claims need to be cleaned up, etc.
```